test/distributed/pipelining/test_schedule.py

```diff
@@ -344,7 +362,190 @@ def test_grad_with_manual_interleaved(self, ScheduleClass):
 
 instantiate_parametrized_tests(ScheduleTest)
 
+
+def format_pipeline_order(pipeline_order: Dict[int, List[Optional[_Action]]]):
+    import itertools
+
+    column_labels = ["Step " + str(i) for i in range(len(pipeline_order[0]))]
```

Comment:
nit: 2 changes that would make this a lot more readable
- transpose, so you have up to RANKS cols and STEPS rows, since RANKS<<STEPS
- implement __repr__ for Action or generally, format such that you use one letter + # per item (like F1 for forward microbatch1)

Reply:
good suggestions, will add. I was also planning on moving this out to its own utility and also printing this out if there is an error

Reply:
now looks like this:

![nun local stages=2 nun microbatches=8 group size=8](https://github.com/pytorch/pytorch/assets/14858254/27339157-697b-47a7-a4a4-9975b77c3597)